LoongArch-ELF-ABI-EN.adoc

```diff
@@ -0,0 +1,206 @@
+= LoongArch ELF ABI specification
+Loongson Technology Corporation Limited
+V1.00
+
+:encoding: UTF-8
+:frame: top
+:grid: rows
+:valign: middle
+
+== Register Convention
+.Integer Register Convention
+[width="80%",options="header",cols="2m,2m,^5,^3"]
+|====================================================================
+|Name     |Alias        |Meaning             |Preserved across calls
+|$r0      |$zero        |Constant zero       |--
+|$r1      |$ra          |Return address      |No
+|$r2      |$tp          |Thread pointer      |--(Unallocatable)
+|$r3      |$sp          |Stack pointer       |Yes
+|$r4-$r11 |$a0-$a7      |Argument registers  |No
+|$r4-$r5  |$v0-$v1      |Return value        |No
+|$r12-$r20|$t0-$t8      |Temp registers      |No
+|$r21     |$x           |Reserved            |--(Unallocatable)
+|$r22     |$fp          |Frame pointer       |Yes
+|$r23-$r31|$s0-$s8      |Subroutine register variables   |Yes
+|====================================================================
+
+.Floating-point Register Convention
+[width="80%",options="header",cols="2m,2m,^5,^3"]
+|====================================================================
+|Name     |Alias        |Meaning            |Preserved across calls
+|$f0-$f7  |$fa0-$fa7    |Argument registers |No
+|$f0-$f1  |$fv0-$fv7    |Return value       |No
+|$f8-$f23 |$ft0-$ft15   |Temp registers     |No
+|$f24-$f31|$fs0-$fs7    |Subroutine register variables   |Yes
+|====================================================================
+
+
+== Type Size and Alignment
+
+.LP64 ABI
+[width="80%",options="header",cols="2m,^5,^3"]
+|====================================================================
+|Scalar type               |Size |Alignment
+|unsigned/signed char      |1    |1
```

Comment:
What's the signedness of bare `char`? RISC-V and AArch64 both have it `unsigned`, MIPS and x86 have it `signed`. According to early toolchain binaries it seems to be signed, but I personally dislike being different from the "new world" conventions.